src/go/k8s/pkg/resources/configuration/patch.go

```diff
@@ -68,10 +70,39 @@ func FourWayMerge(
 	return patch
 }
 
-func valueEquals(v1, v2 interface{}) bool {
-	// TODO verify if there's a better way to handle equality
-	// Problems are e.g. strings and integers, or numeric values becoming float64 when unmarshalled from JSON
-	sv1 := fmt.Sprintf("%v", v1)
-	sv2 := fmt.Sprintf("%v", v2)
+// looseEqual try to determine if two given values are equal using loose comparison.
+// Some example of problems:
+// - Integers and their related string representations should be considered equal
+// - Floating point values should match their integer or string representations if close to an integer number
+// - JSON Number objects (e.g. obtained from JSON APIs) should match their string, integer or float representations
+func looseEqual(v1, v2 interface{}) bool {
```

Comment:
The schema defines a type for each property, so you could normalize all the values via that type knowledge rather than doing a totally general comparison.  e.g. if a field is of type `number` then you can trust the value that the config API gives you will be a floating point, and parse your user input into a floating point before doing the comparison.

Reply:
This is a good point. For total correctness, I may need to change the rpk config interface to deserialize numbers to json.Number (a.k.a. strings), otherwise it will translate integers to float64. I did it in the draft but I reverted it because of multiple failing tests (in core).